src/tools/interop/idt/capture/platform/android/android.py

```diff
@@ -0,0 +1,220 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import ipaddress
+import os
+import time
+import typing
+
+from capture.base import PlatformLogStreamer
+from capture.file_utils import create_standard_log_name
+from capture.shell_utils import BashRunner
+
+
+class Android(PlatformLogStreamer):
+    """
+    Class that supports:
+    - Running synchronous adb commands
+    - Maintaining a singleton logcat stream
+    - Maintaining a singleton screen recording
+    """
+
+    def __init__(self, artifact_dir: str) -> None:
```

Comment:
Synchronizing the clock across the different systems can be really useful. Also, enabling "show touches" is helpful. Maybe we should be requiring the testers to use specific phone models and configure the developer settings in a specific configuration

Reply:
For linux / android, they're likely as tightly sync'd as they can be since they're both full NTP stacks. Unless you're using a direct drive system, getting these more in sync is probably more trouble than its worth, especially since the tightness of the wall clock sync is dwarfed by the log buffer lag. For the DUTs - most don't have a wall clock or a time cluster, The sync accuracy there is unlikely to happen.